packages/react/src/declarative/FeatureFlag.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { useFlag } from '../evaluation';
+import type { FlagQuery } from '../query';
+
+/**
+ * Props for the Feature component that conditionally renders content based on feature flag state.
+ * @interface FeatureProps
+ */
+interface FeatureProps {
+  /**
+   * The key of the feature flag to evaluate.
+   */
+  featureKey: string;
```

Comment:
I think this should be called `key`... or possibly `flagKey`, but I think that's redundant given the name of the component.